pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -184,25 +183,15 @@ func (c *Cluster) Start() error {
 				return fmt.Errorf("failed to update mgr deployment %s. %+v", resourceName, err)
 			}
 		}
+	}
 
-		if err := c.configureOrchestratorModules(); err != nil {
-			logger.Errorf("failed to enable orchestrator modules. %+v", err)
-		}
-
-		if err := c.enablePrometheusModule(c.Namespace); err != nil {
-			logger.Errorf("failed to enable mgr prometheus module. %+v", err)
-		}
-
-		if err := c.configureDashboard(mgrConfig); err != nil {
-			logger.Errorf("failed to enable mgr dashboard. %+v", err)
-		}
-
-		if err := c.configureMgrModules(); err != nil {
-			logger.Errorf("failed to enable mgr module(s) from the spec. %+v", err)
-		}
-
+	if err := c.configureDashboardService(); err != nil {
+		logger.Errorf("failed to enable dashboard. %+v", err)
 	}
 
+	// configure the mgr modules asynchronously so the core orchestration can continue in parallel
```

Comment:
This comment is not true anymore.

Reply:
good catch

Reply:
So this PR is *not* configuring the modules asynchronously? I'm confused.

Reply:
All the modules are configured asynchronously in several goroutines, but we wait for all the goroutines to complete before continuing from this method.